tests/test_core_network.py

```diff
@@ -2717,24 +2726,154 @@ def test_generate_validation_report_with_correct_schedule(correct_schedule):
     assert_semantically_equal(report, correct_report)
 
 
+def test_long_links_show_up_in_validation_report():
+    n = Network('epsg:27700')
+    n.add_link('1', 1, 2, attribs={'length': 10000, 'capacity': 1, 'freespeed': 1, "modes": ['car', 'bus']})
+    n.add_link('2', 2, 3, attribs={'length': 2, 'capacity': 1, 'freespeed': 2, "modes": ['car', 'bus']})
+
+    report = n.generate_validation_report()
+
+    assert_semantically_equal(
+        report['graph']['link_attributes']['links_over_1000_length'],
```

Comment:
Yes! This is exactly what I'm talking about! The assertion here is narrow enough to protect the test from changes in bits of the code that it does not care about.